File: package/umt_wasm/tests/math/test_deg_to_rad.rs
use umtn_wasm::deg_to_rad;

#[test]
fn test_deg_to_rad() {
    assert_eq!(deg_to_rad(0.0), 0.0);
    assert_eq!(deg_to_rad(90.0), 1.5707963267948966);
    assert_eq!(deg_to_rad(180.0), 3.141592653589793);
    assert_eq!(deg_to_rad(270.0), 4.71238898038469);
    assert_eq!(deg_to_rad(360.0), 6.283185307179586);
}

File: package/umt_python/tests/unit/math/test_deg_to_rad.py
import math
import unittest

from src.math import deg_to_rad


class TestDegToRad(unittest.TestCase):
    def test_basic_cases(self):
        self.assertAlmostEqual(deg_to_rad(180), math.pi)
        self.assertAlmostEqual(deg_to_rad(90), math.pi / 2)
        self.assertAlmostEqual(deg_to_rad(0), 0)

    def test_edge_cases(self):
        self.assertAlmostEqual(deg_to_rad(360), 2 * math.pi)
        self.assertAlmostEqual(deg_to_rad(-90), -math.pi / 2)

    def test_docstring_example(self):
        self.assertAlmostEqual(round(deg_to_rad(180), 10), 3.1415926536)


if __name__ == "__main__":
    unittest.main()
